cmake/kokkos_arch.cmake

```diff
@@ -222,6 +223,14 @@ IF (KOKKOS_ARCH_EPYC)
   )
 ENDIF()
 
+IF (KOKKOS_ARCH_GFX906)
+  SET(KOKKOS_ARCH_AMD_GFX906 ON)
+  COMPILER_SPECIFIC_LINKER_FLAGS(
+	LINK_OPTIONS
+        DEFAULT --amdgpu-target=gfx906
+  )
+ENDIF() 
```

Comment:
We need to do this correct based on which backends are enabled and which compiler. This architecture will be relevant both for HIP and OpenMPTarget. I have a branch on my system to add some of this correctly.